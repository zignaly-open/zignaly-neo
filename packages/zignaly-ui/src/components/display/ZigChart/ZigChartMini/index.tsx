import React from "react";
import { VictoryArea, VictoryGroup, VictoryLine } from "victory";
import { ChartLayoutMini } from "../styles";
import { ChartMiniProps } from "../types";
import { useChartData } from "../hooks";
import GraphColors from "../GraphColors";

const ZigChartMini = ({ data, midLine, height }: ChartMiniProps) => {
  const { data: processedData, color, gradient } = useChartData(data);

  return (
    <ChartLayoutMini height={height}>
      <GraphColors />
      <VictoryGroup padding={{ top: 5, bottom: 10 }}>
        <VictoryArea
          style={{
            data: {
              fill: `url(#${gradient})`,
              strokeWidth: 4,
              stroke: color,
            },
          }}
          data={processedData}
        />
        {midLine && (
          <VictoryLine
            style={{
              data: {
                stroke: "grey",
                strokeDasharray: 6,
                strokeWidth: 2,
                strokeOpacity: 0.7,
              },
            }}
            data={[
              { x: 0, y: 0 },
              { x: processedData.length, y: 0 },
            ]}
          />
        )}
      </VictoryGroup>
    </ChartLayoutMini>
  );
};

export default ZigChartMini;
