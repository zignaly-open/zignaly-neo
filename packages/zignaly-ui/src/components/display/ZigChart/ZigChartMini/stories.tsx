import * as React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import ZigChartMini from "./index";

export default {
  title: "Display/Charts/ZigChartMini",
  component: ZigChartMini,
  argTypes: {
    midLine: {
      control: "radio",
      options: { on: true, off: false },
    },
    height: {
      control: "number",
    },
  },
} as ComponentMeta<typeof ZigChartMini>;

const Template: ComponentStory<typeof ZigChartMini> = (args) => <ZigChartMini {...args} />;

export const ExampleChart = Template.bind({});
ExampleChart.args = {
  data: [
    { x: "Jul 1", y: 10 },
    { x: "Jul 2", y: 15 },
    { x: "Jul 3", y: 23 },
    { x: "Jul 4", y: 15 },
    { x: "Jul 5", y: 17 },
    { x: "Jul 6", y: 20 },
    { x: "Jul 7", y: 25 },
  ],
  height: 150,
  midLine: true,
};

export const ExampleChart2 = Template.bind({});
ExampleChart2.args = {
  data: [
    { x: "Jul 1", y: 10 },
    { x: "Jul 2", y: 15 },
    { x: "Jul 3", y: 23 },
    { x: "Jul 4", y: 15 },
    { x: "Jul 5", y: -17 },
    { x: "Jul 6", y: 20 },
    { x: "Jul 7", y: -25 },
  ],
  midLine: true,
  height: 100,
};
