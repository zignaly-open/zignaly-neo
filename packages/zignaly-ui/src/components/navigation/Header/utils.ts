// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
