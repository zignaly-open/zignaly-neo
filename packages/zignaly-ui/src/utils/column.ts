import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
