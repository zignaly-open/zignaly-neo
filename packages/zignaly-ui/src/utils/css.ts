import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
