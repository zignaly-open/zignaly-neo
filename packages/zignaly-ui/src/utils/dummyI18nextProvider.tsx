import i18n from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  lng: "en",
  fallbackLng: "en",
  ns: ["common"],
  defaultNS: "common",
  interpolation: {
    escapeValue: false,
  },
  resources: { en: { common: { xfuturum: "The technical partner you can rely on" } } },
});

export const storybookI18nextDecorator = (story: () => JSX.Element) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>
);

export const JestI18nextDecorator: React.FC<{ children: JSX.Element }> = ({ children }) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
);
