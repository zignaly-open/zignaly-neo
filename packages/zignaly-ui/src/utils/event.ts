export const changeEvent = (name: string | undefined, value: any) => {
  return {
    target: {
      value,
      name,
    },
    type: "change",
  };
};
