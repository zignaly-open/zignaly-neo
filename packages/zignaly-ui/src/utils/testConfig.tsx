import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
