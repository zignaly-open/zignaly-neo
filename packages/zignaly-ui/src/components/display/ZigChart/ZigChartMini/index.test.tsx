import React from "react";
import ZigChartMini from ".";
import { renderWithProvidersUi } from "../../../../utils/testConfig";
import { Box } from "@mui/system";

const mockChartData = [10, 20, 25, 15, 35];

describe("ZigChartMini component tests", () => {
  it("renders ZigChartMini component with data", () => {
    const { container } = renderWithProvidersUi(
      <Box width={"500px"}>
        <ZigChartMini id="test" data={mockChartData} midLine height={300} width={400} />
      </Box>,
    );

    expect(container.querySelector("#test")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
