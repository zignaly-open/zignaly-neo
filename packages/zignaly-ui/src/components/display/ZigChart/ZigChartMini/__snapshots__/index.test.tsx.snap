// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZigChartMini component tests renders ZigChartMini component with data 1`] = `
<div>
  <div
    class="MuiBox-root css-qxh7fn"
  >
    <div
      class="css-1ryu3yr"
      height="300"
      id="test"
    >
      <div
        class="VictoryContainer"
        style="height: 100%; width: 100%; user-select: none; pointer-events: none; position: relative;"
      >
        <svg
          height="300"
          role="img"
          style="pointer-events: all; width: 100%; height: 100%;"
          viewBox="0 0 400 300"
          width="400"
        >
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: transparent; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="0"
              x2="400"
              y1="288.99999999999994"
              y2="288.99999999999994"
            />
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-0"
                x="100"
                y="311.9699999999999"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="fill: transparent; font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent;"
                  text-anchor="middle"
                  x="100"
                >
                  1.0
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-1"
                x="200"
                y="311.9699999999999"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="fill: transparent; font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent;"
                  text-anchor="middle"
                  x="200"
                >
                  2.0
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-2"
                x="300"
                y="311.9699999999999"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="fill: transparent; font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent;"
                  text-anchor="middle"
                  x="300"
                >
                  3.0
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-3"
                x="400"
                y="311.9699999999999"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="fill: transparent; font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent;"
                  text-anchor="middle"
                  x="400"
                >
                  4.0
                </tspan>
              </text>
            </g>
          </g>
          <g
            clip-path="url(#victory-clip-2)"
            style="height: 100%; width: 100%; user-select: none;"
          >
            <defs>
              <clippath
                id="victory-clip-2"
              >
                <rect
                  height="285"
                  vector-effect="non-scaling-stroke"
                  width="400"
                  x="0"
                  y="5"
                />
              </clippath>
            </defs>
            <g>
              <path
                d="M0,208.14285714285714C33.333333333333336,177.82142857142856,66.66666666666666,147.49999999999997,100,127.28571428571426C133.33333333333334,107.07142857142856,166.66666666666666,86.85714285714285,200,86.85714285714285C233.33333333333334,86.85714285714285,266.6666666666667,167.7142857142857,300,167.7142857142857C333.3333333333333,167.7142857142857,366.6666666666667,86.85714285714286,400,6.000000000000003L400,288.99999999999994C366.6666666666667,288.99999999999994,333.3333333333333,288.99999999999994,300,288.99999999999994C266.6666666666667,288.99999999999994,233.33333333333334,288.99999999999994,200,288.99999999999994C166.66666666666666,288.99999999999994,133.33333333333334,288.99999999999994,100,288.99999999999994C66.66666666666666,288.99999999999994,33.333333333333336,288.99999999999994,0,288.99999999999994Z"
                role="presentation"
                shape-rendering="auto"
                style="fill: url(#chart-gradient-mini-green); stroke-width: 2; stroke: none;"
              />
              <path
                d="M0,208.14285714285714C33.333333333333336,177.82142857142856,66.66666666666666,147.49999999999997,100,127.28571428571426C133.33333333333334,107.07142857142856,166.66666666666666,86.85714285714285,200,86.85714285714285C233.33333333333334,86.85714285714285,266.6666666666667,167.7142857142857,300,167.7142857142857C333.3333333333333,167.7142857142857,366.6666666666667,86.85714285714286,400,6.000000000000003"
                role="presentation"
                shape-rendering="auto"
                style="fill: none; stroke-width: 2; stroke: #039179;"
              />
            </g>
          </g>
          <g
            clip-path="url(#victory-clip-3)"
            style="height: 100%; width: 100%; user-select: none;"
          >
            <defs>
              <clippath
                id="victory-clip-3"
              >
                <rect
                  height="285"
                  vector-effect="non-scaling-stroke"
                  width="400"
                  x="0"
                  y="5"
                />
              </clippath>
            </defs>
            <path
              d="M0,288.99999999999994L400,288.99999999999994"
              role="presentation"
              shape-rendering="auto"
              style="fill: transparent; stroke: grey; stroke-dasharray: 6; stroke-width: 2; stroke-opacity: 0.7;"
            />
          </g>
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        >
          <svg
            height="300"
            style="overflow: visible; width: 100%; height: 100%;"
            viewBox="0 0 400 300"
            width="400"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
